CMakeLists.txt

```
@@ -161,11 +161,21 @@ function(packageProject)
     COMPONENT "${PROJECT_NAME}_Development"
   )
 
-  install(
-    DIRECTORY ${PROJECT_INCLUDE_DIR}/
-    DESTINATION ${PROJECT_INCLUDE_DESTINATION}
-    COMPONENT "${PROJECT_NAME}_Development"
-  )
+  if(DEFINED PROJECT_INCLUDE_HEADER_PATTERN)
+    install(
+      DIRECTORY ${PROJECT_INCLUDE_DIR}/
+      DESTINATION ${PROJECT_INCLUDE_DESTINATION}
+      COMPONENT "${PROJECT_NAME}_Development"
+      FILES_MATCHING
+      PATTERN ${PROJECT_INCLUDE_HEADER_PATTERN}
+    )
+  else()
+    install(
+      DIRECTORY ${PROJECT_INCLUDE_DIR}/
+      DESTINATION ${PROJECT_INCLUDE_DESTINATION}
+      COMPONENT "${PROJECT_NAME}_Development"
+    )
+  endif()
```

I'm not sure if this will work, but maybe we could simplify the code using a default pattern in case one isn't defined?

```suggestion
  if(NOT DEFINED PROJECT_INCLUDE_HEADER_PATTERN)
    set(PROJECT_INCLUDE_HEADER_PATTERN "*")
  endif()
  
  install(
    DIRECTORY ${PROJECT_INCLUDE_DIR}/
    DESTINATION ${PROJECT_INCLUDE_DESTINATION}
    COMPONENT "${PROJECT_NAME}_Development"
    FILES_MATCHING
    PATTERN "${PROJECT_INCLUDE_HEADER_PATTERN}"
  )
```

---

not needed, the default is OK, install all under `include` directory

---

Sure, it should definitely have the same behaviour. The only advantage would be using slightly less lines of code and being easier to read imo.